src/config.rs

```diff
@@ -0,0 +1,136 @@
+//! Provides a simple way of reading configuration files.
+//!
+//! Reads a TOML config file for the IPC Agent and deserializes it in a type-safe way into a
+//! [`Config`] struct. Usage:
+//!
+//! ```
+//! let c: Config = Config::from_file("config/example.toml");
+//! ```
+
+use std::collections::HashMap;
+use std::fmt::Formatter;
+use std::fs;
+use std::str::FromStr;
+
+use anyhow::Result;
+use fvm_shared::address::Address;
+use ipc_sdk::subnet_id::SubnetID;
+use serde::de::{Error, SeqAccess};
+use serde::{Deserialize, Deserializer};
+use url::Url;
+
+/// The top-level struct representing the config. Calls to [`Config::from_file`] deserialize into this
+/// struct.
+#[derive(Deserialize)]
+pub(crate) struct Config {
+    pub subnets: HashMap<String, Subnet>,
+}
+
+/// Represents a subnet declaration in the config.
+#[derive(Deserialize)]
+pub struct Subnet {
+    #[serde(deserialize_with = "deserialize_path")]
+    path: SubnetID,
+    rpc_api: Url,
+    auth_token: Option<String>,
+    #[serde(deserialize_with = "deserialize_accounts", default)]
+    accounts: Vec<Address>,
```

Comment:
I think would be nice to add a link to the design doc, explaining what `accounts` stand for.

Reply:
I get your point. However, I think we should be adding a link to the design doc in a README.md file. That's because we'll have these domain-specific terms spread out throughout the codebase and we don't want to add a link to each one of them. Wdyt?

Reply:
Yep, that works!

Reply:
The README will be added in a follow-up PR once we have more meat about the ux the agent.